### DISCUSSION TITLE
Zephyr use lot of space

### DISCUSSION THREAD

**Discussion by fab-uleuh** [zephyrproject-rtos/zephyr#77191] [Q&A]
Hello

I've noticed that Zephyr uses a lot of memory with all its libraries. My code only takes up 15%, but I'm already out of available space. I've been trying to play with flags to reduce flash size.  Many of Zephyr's embedded functions aren't used in my project. I thought the compiler would remove unused functions. Am I missing something here? Are there any strategies to optimize Zephyr's memory usage for my specific application?

Thanks 

<img width="1058" alt="Capture d’écran 2024-08-17 à 19 09 31" src="https://github.com/user-attachments/assets/ff6e162a-24b8-4cde-8248-8b62bf7cea98">

**Comment by vChavezB** [comment]
I would try to disable logging, printk and debug optimizations. 

rather than that analyze the report anfd find out what is consuming the most space.

https://docs.zephyrproject.org/latest/develop/optimizations/tools.html

a discussion regarding footprint

https://github.com/zephyrproject-rtos/zephyr/issues/32875
